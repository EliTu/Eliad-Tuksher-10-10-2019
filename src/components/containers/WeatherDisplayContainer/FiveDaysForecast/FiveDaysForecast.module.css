.FiveDaysForecastStyles {
    width: 95%;
    height: 50%;
    margin: 0 auto;
    display: flex;
    justify-content: center;
    align-content: center;
    align-items: center;
}


/* Medium screens */
@media only screen and (max-width: 600px) {

    .FiveDaysForecastStyles {
        flex-direction: column;
        justify-content: none;
        align-content: none;
        align-items: none;
        height: 55%;
    }
}

/* extra Small screens */
@media only screen and (max-width: 350px) {

    .SearchResultsStyles li {
        font-size: var(--text-size-medium);
        height: 30px;
    }
}