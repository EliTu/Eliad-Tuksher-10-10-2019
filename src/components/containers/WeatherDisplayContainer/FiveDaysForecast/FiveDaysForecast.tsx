import React, { memo } from 'react';
import { connect } from 'react-redux';
import Card from '../../../display/UI/Card/Card';
import {
	setCorrectDateFormat,
	setDayOfTheWeek,
} from '../../../../utilities/convert-functions/dates';
import fahrenheitToCelsius from '../../../../utilities/convert-functions/fahrenheitToCelsius';
import styles from './FiveDaysForecast.module.css';

interface IProps {
	forecastResults: {
		EpochDate: number;
		Date: string;
		Temperature: {
			Minimum: { Value: number; Unit: string };
			Maximum: { Value: number; Unit: string };
		};
		Day: { IconPhrase: string };
		Link: string;
	}[];
}

export const FiveDaysForecast: React.FC<IProps> = ({ forecastResults }) => {
	const { FiveDaysForecastStyles } = styles;

	return (
		<div className={FiveDaysForecastStyles}>
			{forecastResults.map(
				({ EpochDate, Date, Temperature, Day, Link }, i) => {
					// Convert EpochDate into a day of the week string:
					let day = setDayOfTheWeek(EpochDate);
					day =
						i === 0
							? `${day} (Today)`
							: i === 1
							? `${day} (Tomorrow)`
							: day;

					const { Value: maxVal } = Temperature.Minimum;
					const maxCelsiusVal: number = fahrenheitToCelsius(maxVal);
					const maxTemp: number = maxCelsiusVal;

					const { Value: minVal } = Temperature.Maximum;
					const minCelsiusVal: number = fahrenheitToCelsius(minVal);
					const minTemp: number = minCelsiusVal;

					// Get the weather text:
					const { IconPhrase } = Day;

					// Get the DD-MM-YYYY date format:
					const date = setCorrectDateFormat(Date);

					return (
						
							<Card
								key={EpochDate}
								mainHeading={day}
								date={date}
								maxValue={maxTemp}
								minValue={minTemp}
								description={IconPhrase}
								link={Link}
							/>
					
					);
				}
			)}
		</div>
	);
};

// Redux setup:
const mapStateToProps = (state: any) => {
	return {
		forecastResults: state.fiveDaysForecast.resultList,
	};
};

export default connect(mapStateToProps)(memo(FiveDaysForecast));
