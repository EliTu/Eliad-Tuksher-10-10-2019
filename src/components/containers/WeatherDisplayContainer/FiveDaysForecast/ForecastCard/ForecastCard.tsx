import React from 'react';
import styles from './Card.module.css';

interface IProps {
	mainHeading: string;
	date: string;
	maxTempData: string;
	minTempData: string;
	description: string;
	link: string;
}

const ForecastCard: React.FC<IProps> = ({
	mainHeading,
	date,
	maxTempData,
	minTempData,
	description,
	link,
}) => {
	const { CardStyles } = styles;

	return (
		<div className={CardStyles}>
			<h3>
				<a href={link}>{mainHeading}</a>
			</h3>
			<h4>{date}</h4>
			<h4>{`${maxTempData} - ${minTempData}`}</h4>
			<p>{description}</p>
		</div>
	);
};

export default ForecastCard;
