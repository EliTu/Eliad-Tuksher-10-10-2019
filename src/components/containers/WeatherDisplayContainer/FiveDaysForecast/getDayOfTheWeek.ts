const getDayOfTheWeek = (epochDate: number) => {
	// const days = [
	// 	'Sunday',
	// 	'Monday',
	// 	'Tuesday',
	// 	'Wednesday',
	// 	'Thursday',
	// 	'Friday',
	// 	'Saturday',
	// ];
	// let d = new Date(year, --month, day);
	// return d && days[d.getDay()];

	const date = new Date(epochDate * 1000);
	const dayOfWeekV = [
		'Sunday',
		'Monday',
		'Tuesday',
		'Wednesday',
		'Thursday',
		'Friday',
		'Saturday',
	];
	const day = dayOfWeekV[date.getDay()];

	return day;
};

export default getDayOfTheWeek;
