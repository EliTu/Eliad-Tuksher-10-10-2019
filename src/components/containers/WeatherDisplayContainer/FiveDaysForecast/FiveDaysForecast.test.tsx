import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { FiveDaysForecast } from './FiveDaysForecast';
import initialShallowRender from '../../../../utilities/test-utilities/initialShallowRender';
import ForecastCard from './ForecastCard/ForecastCard';

describe('FiveDaysWeather component', () => {
	let component: ShallowWrapper;
	beforeEach(
		() =>
			(component = shallow(
				<FiveDaysForecast
					forecastResults={[
						{
							EpochDate: 1571371200,
							Date: '10-05-19',
							Minimum: { Value: 20, Unit: 'C' },
							Maximum: { Value: 30, Unit: 'C' },
							Day: { IconPhrase: 'Warm' },
							Link: 'www',
						},
						{
							EpochDate: 1571457600,
							Date: '10-05-19',
							Minimum: { Value: 20, Unit: 'C' },
							Maximum: { Value: 30, Unit: 'C' },
							Day: { IconPhrase: 'Warm' },
							Link: 'www',
						},
						{
							EpochDate: 1571544000,
							Date: '10-05-19',
							Minimum: { Value: 20, Unit: 'C' },
							Maximum: { Value: 30, Unit: 'C' },
							Day: { IconPhrase: 'Warm' },
							Link: 'www',
						},
						{
							EpochDate: 1571630400,
							Date: '10-05-19',
							Minimum: { Value: 20, Unit: 'C' },
							Maximum: { Value: 30, Unit: 'C' },
							Day: { IconPhrase: 'Warm' },
							Link: 'www',
						},
						{
							EpochDate: 1571716800,
							Date: '10-05-19',
							Minimum: { Value: 20, Unit: 'C' },
							Maximum: { Value: 30, Unit: 'C' },
							Day: { IconPhrase: 'Warm' },
							Link: 'www',
						},
					]}
				/>
			))
	);

	it('should render the component without errors', () => {
		initialShallowRender(component, 'FiveDaysForecastStyles');
	});

	it('should render a ul with a list of 5 Card components', () => {
		const ul = component.find('ul');
		const card = component.find('ForecastCard');

		expect(ul.length).toBe(1);
		expect(card).toBeTruthy();
		expect(card.length).toBe(5);
	});
});
