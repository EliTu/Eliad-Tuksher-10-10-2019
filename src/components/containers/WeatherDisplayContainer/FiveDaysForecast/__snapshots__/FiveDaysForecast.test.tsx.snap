// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiveDaysWeather component should render the component without errors 1`] = `
<div
  className="FiveDaysForecastStyles"
>
  <ul>
    <ForecastCard
      date="10-05-19"
      description="Warm"
      link="www"
      mainHeading="Friday"
      maxTempData="20 C"
      minTempData="30 C"
    />
    <ForecastCard
      date="10-05-19"
      description="Warm"
      link="www"
      mainHeading="Saturday"
      maxTempData="20 C"
      minTempData="30 C"
    />
    <ForecastCard
      date="10-05-19"
      description="Warm"
      link="www"
      mainHeading="Sunday"
      maxTempData="20 C"
      minTempData="30 C"
    />
    <ForecastCard
      date="10-05-19"
      description="Warm"
      link="www"
      mainHeading="Monday"
      maxTempData="20 C"
      minTempData="30 C"
    />
    <ForecastCard
      date="10-05-19"
      description="Warm"
      link="www"
      mainHeading="Tuesday"
      maxTempData="20 C"
      minTempData="30 C"
    />
  </ul>
</div>
`;
